nemo/nemo/backends/pytorch/actions.py

```
@@ -997,6 +1000,132 @@ def _get_all_modules(
+                if input_example is None:
+                    # Route 1 - via torch.jit.script
+                    traced_m = torch.jit.script(module)
+                    traced_m.save(output)
+                else:
+                    # Route 2 - via tracing
+                    traced_m = torch.jit.trace(module, input_example)
+                    traced_m.save(output)
+            elif d_format == DeploymentFormat.ONNX:
+                if input_example is None:
+                    raise ValueError(
+                        f'Example input is None, but ONNX tracing was'
+                        f' attempted')
+                torch.onnx.export(module, input_example, output,
+                                  input_names=input_names,
+                                  output_names=output_names,
+                                  verbose=True,
+                                  export_params=True,
+                                  do_constant_folding=True,
+                                  dynamic_axes=dynamic_axes,
+                                  opset_version=10)
+                # fn = output + ".readable"
+                # with open(fn, 'w') as f:
+                #     # Write human-readable graph representation to file as
+                #     well.
+                #     tempModel = onnx.load(output)
+                #     onnx.save(tempModel, output + ".copy")
+                #     onnx.checker.check_model(tempModel)
+                #     pgraph = onnx.helper.printable_graph(tempModel.graph)
+                #     f.write(pgraph)
+
+            elif d_format == DeploymentFormat.PYTORCH:
+                torch.save(module.state_dict(), output)
+                with open(output + ".json", 'w') as outfile:
+                    json.dump(local_parameters, outfile)
+
+            else:
+                raise NotImplemented(
+                    f"Not supported deployment format: {d_format}")
+        except:  # nopep8
```

Does this fail silently?

---

changed - now prints error